stdlib.sh

```diff
@@ -61,6 +61,113 @@ puts_verbose() {
   fi
 }
 
+# Key Value Store Utility
+# -------------------
+
+# This is a utility module that allows you to create a file that serves
+# as a key-value store that can be persisted to disk.
+#
+# It works around using lines formatted like:
+# 
+# key=value
+#
+# and using grep when searching for keys to find the last entry. This 
+# allows an existing key to be overwritten.
+#
+# This is inspired by the "world's simplest database" in Ch 3 of
+# Designing Data-Intensive Applications https://dataintensive.net/
+
+# Ensure that the file that we would like to persist this store to
+# exists. Also handles creating the containing directory if it doesn't
+# exist yet. 
+#
+# Will not overwrite the store if it already exists
+# Usage: kv_create file
+kv_create() {
+  local f=$1
+  mkdir -p "$(dirname "$f")"
+  touch "$f"
+}
+
+# Delete all keys and values from the store
+# Usage: kv_clear file
+kv_clear() {
+  local f=$1
+  echo "" > "$f"
+}
+
+# Set value for a key
+# Usage: kv_set file key value
+kv_set() {
+  if [[ $# -eq 3 ]]; then
```

Comment:
should this raise an error if args incorrect?

Reply:
also, you can make it `if (( $# == 3 )); then`, or, if you want to handle errors, `if (( $# != 3 )); then`, and handle and throw the error in there, so the rest of the function isn't indented

Reply:
in general, as a style, I prefer an `if` with the negated condition that just does a `return` (or throws error or whatever) over wrapping the entire function block in the `if`